db/model/aspect.js

```diff
@@ -303,17 +303,14 @@ module.exports = function aspect(seq, dataTypes) {
              */
             redisOps.deleteKeys(sampleType, aspectType, inst.name);
           } else if (inst.changed('isPublished')) {
-            if (inst.isPublished) {
-              // Prevent any changes to original inst dataValues object
-              const instDataObj = JSON.parse(JSON.stringify(inst.get()));
 
+            // Prevent any changes to original inst dataValues object
+            const instDataObj = JSON.parse(JSON.stringify(inst.get()));
+            redisOps.hmSet(aspectType, inst.name, instDataObj);
+            if (inst.isPublished) {
               redisOps.addKey(aspectType, inst.name);
```

Comment:
the aspect has to be added to the aspect master list even if it is not published.

Reply:
ok will add.